core/src/clients/git.rs

```diff
@@ -611,6 +613,7 @@ impl Git {
         }
     }
 
+    #[instrument(name = "git::run_and_collect_output_internal", err)]
```

Comment:
this `err` will fire an error event when any `Result<T, E>` returns an err so long as `E` implements `fmt::Display`, which I believe `anyhow` does. 

I implemented the `err` here so we don't have to instrument it at any of the call sites (which would also duplicate a bunch of sent data i think)

Reply:
Yeah, the `std::error::Error` trait [requires it](https://doc.rust-lang.org/std/error/trait.Error.html), so anything that's an error should impl Display. :+1: